outlook/src/taskpane/components/Company/CompanySection/CompanySection.tsx

```diff
@@ -106,50 +107,64 @@ class CompanySection extends React.Component<CompanySectionProps, CompanySection
+        if (!this.props.partner.email)
         {
-            let enrichAndCreate = null;
-            if (this.props.partner.isAddedToDatabase() && !this.enriched)
+            let enrichmentInfo = new EnrichmentInfo(EnrichmentInfoType.EnrichContactWithNoEmail,
+                _t("This contact has no email address, no company could be enriched."));
+            this.context.showTopBarMessage(enrichmentInfo);
+            return;
+        }
+
+        this.setState({isLoading: true});
+
+
+        const enrichRequest = sendHttpRequest(HttpVerb.POST, api.baseURL + api.enrichAndUpdateCompany,
+            ContentType.Json, this.context.getConnectionToken(), {
+                partner_id: this.props.partner.id,
+            }, true);
+
+        this.context.addRequestCanceller(enrichRequest.cancel);
+
+        enrichRequest.promise.then((response) => {
+            const parsed = JSON.parse(response);
+
+            if (parsed.result['enrichment_info'])
             {
-                enrichAndCreate = (
-                    <div>
-                        <div>
-                            {_t("No company attached to this contact")}
-                        </div>
-                        <div className="odoo-secondary-button" style={{margin: "8px auto auto auto"}} onClick={this.enrichCompanyForPartnerRequest}>
-                            {_t("Create a Company")}
-                        </div>
-                    </div>
-                );
+                const enrichmentInfo = new EnrichmentInfo(parsed.result['enrichment_info'].type,
+                    parsed.result['enrichment_info'].info);
+                this.context.showTopBarMessage(enrichmentInfo);
+
+                if (enrichmentInfo.type == EnrichmentInfoType.NoData)
+                    this.enriched = true;
+            }
+
+            if (parsed.result['company'])
+            {
+                this.enriched = true;
+                const company = Company.fromJSON(parsed.result.company);
+                this.enriched = true;
+                this.setState({isCollapsed: false, company: company, isLoading: false});
+                this.props.onCompanyUpdated();
             }
             else
             {
-                if (!this.context.isConnected())
-                {
-                    enrichAndCreate = (
-                        <div>
-                            {_t("No company linked to this contact could be enriched")}
-                        </div>
-                    );
-                }
-                else
-                {
-                    enrichAndCreate = (
-                        <div>
-                            {_t("No company linked to this contact could be enriched or found in Odoo")}
-                        </div>
-                    );
-                }
+                this.setState({isCollapsed: false, isLoading: false});
             }
+        }).catch(error => {
+            console.log(error);
```

Comment:
This one is not setting "this.enriched = true", is that intended?
Will that allow the user to retry maybe?

Reply:
Yes exactly, the purpose of not doing this.enriched = true here, is to do not hide the button and allow the user to retry.